mlrun/runtimes/pod.py

```diff
@@ -633,6 +900,25 @@ def with_priority_class(self, name: typing.Optional[str] = None):
             raise mlrun.errors.MLRunInvalidArgumentError(message)
         self.spec.priority_class_name = name
 
+    def with_preemption_mode(self, mode: str):
+        """
+        Preemption modes enable users to run function pods on preemptible nodes, when filled,
+        tolerations, node labels and affinity would be populated correspondingly to the function spec on server side.
+
+        currently supports 3 modes:
+
+        allow - Allows function to be scheduled on preemptible nodes
+        constrains - Makes function to run only on preemptible nodes
+        prevent - Prevents function to be scheduled on preemptible nodes
+
+        :param mode: accepts allow | constraint | prevent defined in :py:class:`~mlrun.api.schemas.PreemptionModes`
```

Comment:
You should accept (express in type hint as well) both a string or an instance of the enum (in which case no need to do the check, but need to transform back to string before assignment)

Reply:
why back to string, we still have all the behaviors we expect from string, even when leaving it as `PreemptibleModes.allow`